tool/firebase_util.py

```diff
@@ -1,14 +1,67 @@
+import time
 
 import firebase_admin
 from firebase_admin import credentials
 from firebase_admin import firestore
 from katikati_pylib.logging import logging
 
+log = None
+firebase_client = None
+
 def init_firebase_client(CRYPTO_TOKEN_PATH):
-    log = logging.Logger(__file__, CRYPTO_TOKEN_PATH)
+    global log
+    if log is None:
+        log = logging.Logger(__file__, CRYPTO_TOKEN_PATH)
     log.info("Setting up Firebase client")
     firebase_cred = credentials.Certificate(CRYPTO_TOKEN_PATH)
     firebase_admin.initialize_app(firebase_cred)
-    firebase_client = firestore.client()
+    global firebase_client
+    if firebase_client is None:
+        firebase_client = firestore.client()
     log.info("Done")
     return firebase_client
+
+def push_collection_to_firestore(collection_root, documents):
+    log.info (f"push_collection_to_firestore {collection_root}")
+    col = firebase_client.collection(collection_root)
+    time_start = time.perf_counter_ns()
+    for document_dict in documents:
+        firestore_format_document_dict = _convert_to_firestore_format(collection_root, document_dict)
+        push_document_to_firestore(collection_root, firestore_format_document_dict)
```

Comment:
The conversion should happen before the method is called - `push_collection_to_firestore` should get only a single document format

```suggestion
        push_document_to_firestore(collection_root, firestore_format_document_dict)
```